src/cdk/a11y/aria-describer.ts

```diff
@@ -0,0 +1,135 @@
+
+/** Container for all registered messages. */
+let messagesContainer: HTMLElement | null = null;
+
+/**
+ * Utility that creates visually hidden elements with a message content. Useful for elements that
+ * want to use aria-describedby to further describe themselves without adding additional visual
+ * content. Must be provided with as part of a component's injection tree so that it has access
+ * to the Renderer.
+ */
+@Injectable()
+export class AriaDescriber {
+  constructor(private _platform: Platform,
+              @Optional() private _renderer: Renderer2) {
+    if (!_renderer) {
+      throw Error('AriaDescriber must be provided through a component\'s provider decorator ' +
+          'so that it has access to the Angular Renderer');
+    }
+  }
+
+  /**
+   * Adds to the host element an aria-describedby reference to a hidden element that contains
+   * the message. If the same message has already been registered, then it will reuse the created
+   * message element.
+   */
+  addDescription(hostElement: HTMLElement, message: string) {
+    if (!this._platform.isBrowser || !message.trim()) { return; }
+
+    if (!registeredMessages.get(message)) {
+      const messageElement = this._createMessageElement(message);
+      registeredMessages.set(message, {messageElement, hostElements: []});
+    }
+
+    const registeredMessage = registeredMessages.get(message)!;
+    registeredMessage.hostElements.push(hostElement);
+    addAriaReferencedId(hostElement, registeredMessage.messageElement.id, 'aria-describedby');
+  }
+
+  /**
+   * Removes the host element's aria-describedby reference to the message element.
+   */
+  removeDescription(hostElement: HTMLElement, message: string) {
+    if (!this._platform.isBrowser || !message.trim()) { return; }
+
+    const registeredMessage = registeredMessages.get(message)!;
+    registeredMessage.hostElements = registeredMessage.hostElements.filter(el => el != hostElement);
+    removeAriaReferencedId(hostElement, registeredMessage.messageElement.id, 'aria-describedby');
+
+    if (registeredMessage.hostElements.length == 0) {
+      this._deleteMessageElement(message);
+    }
+
+    if (messagesContainer!.childNodes.length == 0) {
+      this._deleteMessagesContainer();
+    }
+  }
+
+  /** Unregisters all created message elements and removes the message container. */
+  _unregisterAllMessages() {
+    registeredMessages.forEach((registeredMessage, message) => {
+      registeredMessage.hostElements.forEach(hostElement => {
+        this.removeDescription(hostElement, message);
+      });
+    });
+
+    registeredMessages.clear();
+  }
+
+  /**
+   * Creates a new element in the visually hidden message container element with the message
+   * as its content.
+   */
+  private _createMessageElement(message: string): HTMLElement {
+    const messageElement = this._renderer.createElement('div');
+    this._renderer.setAttribute(messageElement, 'id', `cdk-aria-describedby-message-${nextId++}`);
```

Comment:
Went with `cdk-describedby-##` for the messages, and `cdk-described` for the elements that are described.